src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -372,6 +374,33 @@ public static function underscore($id)
         return strtolower(preg_replace(array('/([A-Z]+)([A-Z][a-z])/', '/([a-z\d])([A-Z])/'), array('\\1_\\2', '\\1_\\2'), str_replace('_', '.', $id)));
     }
 
+    /**
+     * Fetches a variable from the environment.
+     *
+     * @param string The name of the environment variable
+     *
+     * @return scalar The value to use for the provided environment variable name
+     *
+     * @throws EnvNotFoundException When the environment variable is not found and has no default value
+     */
+    protected function getEnv($name)
```

Comment:
On the contrary, in my mind, the ability to put your defaults in a regular parameter section (or in a parameters.yml) is very elegant, and one of the stronger advantages of this approach compared to earlier attempts at this.